### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by SherinV**
**Related Issue:**  open-cluster-management/backlog#9582

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+make test
+```
+
+## Build images
+
+Make sure your code build passed.
+
+```shell
+export BUILD_LOCALLY=1
+make
```
Comment (SherinV) [inline CONTRIBUTING.md:58]: Please review if the command needs to be updated

Comment (jlpadilla) [reply CONTRIBUTING.md:58]: What name are other projects using for the `make <target>`?

With my recent changes we can build a local image with:
```
docker build -f Dockerfile . -t search-collector
```

So we just need to add that to the Makefile

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Repo maintainers can assign you an issue or pull request by leaving a
+`/assign <your Github ID>` comment on the issue or pull request.
+
+## Pre-check before submitting a PR
+
+After your PR is ready to commit, please run following commands to check your code.
+
+```shell
+make test
```
Comment (jlpadilla) [inline CONTRIBUTING.md:49]: Should we add `make lint` here too?

File: CONTRIBUTING.md (hunk 1 above)
Comment (SherinV) [reply CONTRIBUTING.md:58]: I see `make docker-build ` in https://github.com/open-cluster-management/discovery/pull/39/files
and `make image` in https://github.com/open-cluster-management/multicloudhub-repo/pull/368/files

Comment (jlpadilla) [reply CONTRIBUTING.md:59]: Our search-operator also has `make docker-build`.  Most likely it comes from the operator SDK, so let's stay consistent with that.

**Review by jlpadilla — APPROVED** [review]

**Pull request merged**